src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -164,6 +181,55 @@ def _list_services(self, cmd):
 
             return HandleCommandResult(odata="\n".join(lines))
 
+    def _create_osd(self, cmd):
```

Comment:
Please have a look at 

https://github.com/ceph/ceph/blob/c7940db6b65b86d758606fa4cb1f27839df09bc0/src/pybind/mgr/orchestrator_cli/module.py#L197-L213

Could you extract, reuse and extend that code? Please make sure you're not breaking Rook here.

Reply:
I have take advantage of this modification and i have implemented the "Drive group" class in the Orchestrator. Now Orchestrator cli.create_osds uses a "drive_group" object that is passed  to the internal orchestrator module. Each internal orchestrator module process the "drive_group" object as required. I think that when we will have teh ceph-volume "implementation" probably this object will be passed directly. In any case, now we have in the orchestrator a "drive group" object to manage any kind of transformation/process.

Reply:
I can't test the Rook operator easily.... i think that the modifications i have done are quite inoffensives, but it would be nice if somebody can do this kind of test. (i will assist as needed)